### PR TITLE
Reset health

### PR DESCRIPTION
**Pull request by GigaDev90**
This PR introduces a new configurable setting (disabled by default) which reset the mana and health of the player before and after Duel.
Thanks to shindarth for help me during the work.

### CODE REVIEW

**Comment by tkrokli** [comment]
You need to update your local branch `GigaDev90:reset-health` so it is even with `TrinityCore:3.3.5`.
Ironically enough, you must include both of @ShinDarth's PRs/commits, because they made changes to the same files as you want to make changes to. (All the other commits of the last 12 are DB updates.) Make sure you perform a `rebase`, not a merge, on your `reset-health` branch.

**Comment by ShinDarth** [comment]
@GigaDev90 

try:
```
git remote add upstream git@github.com:TrinityCore/TrinityCore.git
git fetch upstream
git merge upstream/3.3.5
git push
```

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -7624,6 +7624,21 @@ void Player::DuelComplete(DuelCompleteType type)
             ChatHandler(duel->opponent->GetSession()).PSendSysMessage(LANG_COOLDOWN_NOT_RESET_AFTER_DUEL);
     }
 
+    //need for save and restore health and mana before and after the a duel
+    if (sWorld->getBoolConfig(CONFIG_RESTORE_HEALTHMANA_AFTER_DUEL))
+    {
+        RestoreHealthAfterDuel();
+        duel->opponent->RestoreHealthAfterDuel();
+        //check if player class uses mana
+         if (!(getClass() == CLASS_WARRIOR || getClass() == CLASS_ROGUE ||
```
Comment (Naios) [inline src/server/game/Entities/Player/Player.cpp]:
```c++
if (getPowerType() == POWER_MANA)
    RestoreManaAfterDuel();
```

File: src/server/game/Handlers/DuelHandler.cpp [hunk 2]
```diff
@@ -46,6 +46,30 @@ void WorldSession::HandleDuelAcceptedOpcode(WorldPacket& recvPacket)
...
+    //need for save and restore health and mana before and after the a duel
+    if (sWorld->getBoolConfig(CONFIG_RESTORE_HEALTHMANA_AFTER_DUEL))
+    {
+        player->SaveHealthBeforeDuel();
+        player->SetHealth(player->GetMaxHealth());
+
+        plTarget->SaveHealthBeforeDuel();
+        plTarget->SetHealth(plTarget->GetMaxHealth());
+        //check if player class uses mana
+        if (!(player->getClass() == CLASS_WARRIOR || player->getClass() == CLASS_ROGUE ||
```
Comment (Naios) [inline src/server/game/Handlers/DuelHandler.cpp]: `if (getPowerType() == POWER_MANA)` too

File: src/server/game/World/World.cpp [hunk 3]
```diff
@@ -1191,6 +1191,7 @@ void World::LoadConfigSettings(bool reload)
     m_bool_configs[CONFIG_START_ALL_SPELLS] = sConfigMgr->GetBoolDefault("PlayerStart.AllSpells", false);
     m_int_configs[CONFIG_HONOR_AFTER_DUEL] = sConfigMgr->GetIntDefault("HonorPointsAfterDuel", 0);
     m_bool_configs[CONFIG_RESET_COOLDOWN_AFTER_DUEL] = sConfigMgr->GetBoolDefault("ResetCoolDownAfterDuel", 0);
+    m_bool_configs[CONFIG_RESTORE_HEALTHMANA_AFTER_DUEL] = sConfigMgr->GetBoolDefault("RestoreHealthManaAfterDuel", 0);
```
Comment (Naios) [inline src/server/game/World/World.cpp]:
```c++
GetBoolDefault("RestoreHealthManaAfterDuel", false)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^
```

File: src/server/worldserver/worldserver.conf.dist [hunk 4]
```diff
@@ -2614,6 +2614,14 @@ HonorPointsAfterDuel = 0
 ResetCoolDownAfterDuel = 0
 
 #
+#    RestoreHealthManaAfterDuel
+#        Description: restore heatlh and mana after duel.
```
Comment (Naios) [inline src/server/worldserver/worldserver.conf.dist]: `heatlh ` typo

**Comment by Naios** [comment]
Consider to use just

```sh
git remote add tc git@github.com:TrinityCore/TrinityCore.git
git pull --rebase tc 3.3.5
```

next time

**Comment by ShinDarth** [comment]
https://github.com/TrinityCore/TrinityCore/pull/15728

**Pull request closed without merging**